lib/styles.js

```diff
@@ -340,4 +342,133 @@ module.exports = function (expect) {
             this.sp().commentBlock(pen);
         }
     });
+
+    expect.addStyle('diffFragment', function (text, baseStyle, specialCharStyle, markUpSpecialCharacters) {
+        if (markUpSpecialCharacters) {
+            text.split(specialCharRegExp).forEach(function (part) {
+                if (specialCharRegExp.test(part)) {
+                    this[specialCharStyle || baseStyle](utils.escapeChar(part));
+                } else {
+                    this[baseStyle](part);
+                }
+            }, this);
+        } else {
+            this[baseStyle](text);
+        }
+    });
+
+    expect.addStyle('stringDiff', function (actual, expected, options) {
+        options = options || {};
+        var type = options.type || 'WordsWithSpace';
+        var diffLines = [];
+        var lastPart;
+        stringDiff.diffLines(actual, expected).forEach(function (part) {
```

Comment:
I think we can simplify things if we only used `diffAddedLine`, `diffRemovedLine` with one line and not multiple lines. Then we would be able to move the `-/+` signs to these methods:

```js
    expect.addStyle('diffAddedLine', function (line) {
        this.alt({ text: '+' });
        this.green(line);
    });

    expect.addStyle('diffRemovedLine', function (line) {
        this.alt({ text: '-' });
        this.red(line);
    });

    expect.addStyle('diffUnchangedLine', function (line){
        this.alt({ text: ' ' });
        this.text(line)
    });
```